sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -295,16 +295,21 @@ public virtual PartitionProperties CreatePartitionPropertiesFromResponse(AmqpMes
         {
             AmqpMessage batchEnvelope;
 
+            // If there is a single message, then it will be used as the
+            // envelope.  To avoid mutating the instance and polluting it with
+            // delivery state which prevents it from being sent during retries,
+            // clone it.
+
             if (source.Count == 1)
             {
                 switch (source)
                 {
                     case List<AmqpMessage> messageList:
-                        batchEnvelope = messageList[0];
+                        batchEnvelope = messageList[0].Clone();
```

Comment:
It seems it does only a shallow copy. Is that a problem?

https://github.com/Azure/azure-amqp/blob/master/src/AmqpMessage.cs#L466-L468